WL-4163 Fixed the checkbox for 'use this link as title' in citations.

ouit0408 opened the pull request.

buckett commented on citations-tool/tool/src/webapp/vm/citation/create.vm:
```diff
@@ -376,14 +376,15 @@
+		if( checkBoxClicked.attr( "checked" ) == true )
 		{
 			// clear all checkboxes
 			$( ".prefLink" ).attr( "checked", "" );
```
You can also compact all this and just do: `$(".prefLink").not("#"+checkboxId).removeAttr("checked")` then you don't need to re-add the checked one. Doesn't need updating, just an alternative.

buckett commented:
Always nice to just have the extra sentence explaining why this fixes it, is it because some browsers return false?

ouit0408 commented:
It's because previously we were checking if checked attribute for the checkbox is null or not which could be either true of false, so as per the logic checkbox was not getting unchecked. Nothing to do with the browser.

buckett commented on citations-tool/tool/src/webapp/vm/citation/create.vm:
```diff
@@ -376,14 +376,15 @@
-		if( $( "#" + checkboxId ).attr( "checked" ) != null )
+		var checkBoxClicked = $( "#" + checkboxId );
+		// has this been checked?
+		if( checkBoxClicked.attr( "checked" ) == true )
```
You don't need the `== true` because that's what `if()` does.

ouit0408 replied:
you are right, just checked jquery documentation, pre 1.6 jquery version returns boolean for $( elem ).attr( "checked" ) and on this page we are loading jquery1.4. will correct it

buckett replied:
But what does it return 1.6+ as it may well be upgrade shortly. I think the `jQuery way` of doing this is with the `checkBoxClicked.is(":checked")`.

ouit0408 replied:
I was going to write that as we upgrade the jquery this won't work.Yes the right way is to use element.is(:checked) or element.prop( "checked" ).

ouit0408 committed:
WL-4163 Fixed the checkbox for 'use this link as title'.

In trackPreferredUrl method using 'is(:checked)' method to get the checked
box instead of '.attr(checked)' as the latter one always returns checked
string.

ouit0408 commented:
implemented review comments and have pushed the changes.

buckett commented:
If you see duplicated code like this and you are working on it feel free to extract it out into a common location.
